rust/xaynet-client/src/mobile_client/mod.rs

```diff
@@ -8,26 +8,51 @@ use crate::{
         participant::ParticipantSettings,
     },
 };
+use thiserror::Error;
 use xaynet_core::{
     crypto::{SecretSigningKey, SigningKeyPair},
     mask::Model,
+    InitError,
 };
 
+#[derive(Debug, Error)]
+/// Mobile client errors
+pub enum MobileClientError {
+    #[error("failed to deserialise mobile client: {0}")]
+    /// Failed to deserialise mobile client.
+    DeserialiseErr(#[from] bincode::Error),
```

Comment:
I'd remove the `Err` suffix on the variants. The `Error` is already part of the enum name.

Reply:
Also I think we should follow `serde` convention of using the US the spelling of "(de)seriali*z*ation", since it's the de facto standard in the Rust ecosystem.

Reply:
Yeah you right. I have changed it